core-bundle/src/Image/Studio/LightBoxResult.php

```diff
@@ -0,0 +1,149 @@
+     */
+    public function __construct(ContainerInterface $locator, $filePathOrImage, ?string $url, $sizeConfiguration = null, string $groupIdentifier = null)
+    {
+        if (1 !== \count(array_filter([$filePathOrImage, $url]))) {
+            throw new \InvalidArgumentException('A lightbox must be either constructed with a resource or an url.');
+        }
+
+        $this->locator = $locator;
+        $this->url = $url;
+        $this->groupIdentifier = $groupIdentifier;
+
+        if (null !== $filePathOrImage) {
+            $this->image = $locator
+                ->get(Studio::class)
+                ->createImage($filePathOrImage, $sizeConfiguration ?? $this->getDefaultLightBoxSizeConfiguration())
+            ;
+        }
+    }
+
+    /**
+     * Return if this light box result contains a (resized) image.
+     */
+    public function hasImage(): bool
+    {
+        return null !== $this->image;
+    }
+
+    /**
+     * Return the underlying (resized) image.
+     */
+    public function getImage(): ImageResult
+    {
+        if (!$this->hasImage()) {
+            throw new \RuntimeException('This light box result does not contain a (resized) image.');
+        }
+
+        return $this->image;
+    }
+
+    /**
+     * Return the link url pointing to the resource.
+     */
+    public function getLinkHref(): string
+    {
+        return $this->hasImage() ? $this->image->getImageSrc() : $this->url;
+    }
+
+    /**
+     * Return the light box group identifier.
+     */
+    public function getGroupIdentifier(): string
+    {
+        return $this->groupIdentifier ?? $this->createFallbackGroupIdentifier();
+    }
+
+    /**
+     * Try to get a light box size configuration from the current page's
+     * associated layout. Will return null if not defined or not in a request
+     * context.
+     */
+    private function getDefaultLightBoxSizeConfiguration(): ?array
+    {
+        $framework = $this->locator->get('contao.framework');
+
+        $page = $GLOBALS['objPage'] ?? null;
+
+        if (!$page instanceof PageModel || null === $page->layout) {
+            return null;
+        }
+
+        // Try to get layout
+        /** @var LayoutModel $layoutModelAdapter */
+        $layoutModelAdapter = $framework->getAdapter(LayoutModel::class);
+
+        /** @var LayoutModel|null $layoutModel */
+        $layoutModel = $layoutModelAdapter->findByPk($page->layout);
+
+        if (null === $layoutModel || empty($layoutModel->lightboxSize)) {
+            return null;
+        }
+
+        return StringUtil::deserialize($layoutModel->lightboxSize, true);
+    }
+
+    /**
+     * Create a pseudo-unique identifier. This prevents independent images
+     * being grouped under the same (empty) identifier.
+     */
+    private function createFallbackGroupIdentifier(): string
+    {
+        return substr(md5($this->getLinkHref()), 0, 6);
```

Comment:
I think we should/can get rid of this. In my tests with the default jQuery lightbox, empty lightbox-attributes are not grouped together. Even if this was not the case we should adapt the JavaScript instead I think.

The current implementation now generates the same lightbox id for two images that link to the same lightbox-url, this results in a next-link in the lightbox that shows then the exact same image.

Defaulting to an empty ligthbox-id is the way to go IMO. (Maybe we need to set a lightbox id in the addImageToTemplate method for BC).

Reply:
> The current implementation now generates the same lightbox id for two images that link to the same lightbox-url, this results in a next-link in the lightbox that shows then the exact same image.

True, if we really must generate a fallback lightbox id it should probably be random.

> Defaulting to an empty ligthbox-id is the way to go IMO. (Maybe we need to set a lightbox id in the addImageToTemplate method for BC).

:+1: Do we need to adjust the `moo_mediabox`/`moo_slimbox` js then? These scripts should imo also work without using `Controller::addImageToTemplate`.

btw.: This is where this logic originated from: https://github.com/contao/core/issues/3742

Reply:
Just tested with MooTools, `data-lightbox=""` works fine there too. So I think we can remove this logic altogether.

Reply:
Dropped in 3348c3e7768aaa748346a84c05e06f8bdf784e42.